OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -163,18 +163,36 @@ public void AddSmudge(CPos loc)
 			}
 		}
 
+		public void RemoveSmudge(CPos loc)
+		{
+			if (dirty.ContainsKey(loc))
+			{
+				var tile = dirty[loc];
+				tile.Sprite = null;
+				dirty[loc] = tile;
+			} else {
+				var st = smudges.Keys.Random(world.SharedRandom);
+				dirty[loc] = new Smudge { Type = st, Depth = 0, Sprite = null };
```

Comment:
using null sprite as indicator to update render in this point and remove this null sprite in TickRender function